### PR TITLE
withSavepoint now catches SomeAsyncException

### PR DESCRIPTION
**Pull request by edmundnoble**
`withSavepoint` here is trying to rethrow asynchronous exceptions but fails because it catches `AsyncException` instead of `SomeAsyncException`. I noticed this looking at the logs which show `non-pact exception: AsyncCancelled`.

### CODE REVIEW

File: src/Chainweb/Pact/Backend/Utils.hs [hunk 1]
```diff
@@ -166,7 +166,7 @@ withSavepoint name action = mask $ \resetMask -> do
         liftIO $ evaluate r
     throwErr s = internalError $ "withSavepoint (" <> toText name <> "): " <> s
     handlers = [ Handler $ \(e :: PactException) -> throwErr (sshow e)
-               , Handler $ \(e :: AsyncException) -> throwM e
+               , Handler $ \(e :: SomeAsyncException) -> throwM e
```
Comment (larskuhtz) [inline src/Chainweb/Pact/Backend/Utils.hs:169]: Oh, good catch. Is this change safe to apply or can it break anything? I guess it should be safe, but why may want quickly grep over all use sites.

**Review by larskuhtz — APPROVED** [review]
Approved with one comment

File: src/Chainweb/Pact/Backend/Utils.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/Pact/Backend/Utils.hs:169]: It's used only in two places and it seems fine. Also checked the respective functions in `Chainweb.Utils`, which have correct behavior.

Comment (edmundnoble) [reply src/Chainweb/Pact/Backend/Utils.hs:169]: This should be safe I think, any place which is fine with `ThreadKilled` should also be fine with `AsyncCancelled`.

**Pull request merged**